src/system/SystemObject.h

```diff
@@ -270,17 +259,17 @@ inline void ObjectPool<T, N>::Reset()
 
 /**
  *  @brief
- *      Tries to initially retain the first object in the pool that is not retained by any layer.
+ *      Tries to initially retain the first object in the pool that is not retained.
  */
 template <class T, unsigned int N>
-inline T * ObjectPool<T, N>::TryCreate(Layer & aLayer)
+inline T * ObjectPool<T, N>::TryCreate()
```

Comment:
Should we template enforce that T derives from SystemObject? Or is that not a general requirement?

Reply:
Good suggestion, add (void) static_cast<Object *>(lReturn);  /* In C++-11, this would be a static_assert that T inherits Object. */